stages/org.osbuild.ostree.passwd

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python3
+"""
+Populate buildroot with /etc/passwd and /etc/group from an OSTree checkout
+
+Using the OSTree checkout provided as in input, copy /usr/etc/passwd and
+/usr/lib/passwd, merge them and store the result into /etc/passwd in the
+buildroot. Do the same for /etc/group file.
+
+The use case for this stage is when one wants to preserve UIDs and GIDs
+which might change when the system is build from scratch. Creating these
+files before any RPMs (or other packages) are installed will prevent changes
+in UIDs and GIDs.
+"""
+
+import os
+import shutil
+import sys
+import subprocess
+
+import osbuild.api
+
+
+SCHEMA_2 = """
+"options": {
+  "additionalProperties": false
+},
+"inputs": {
+  "type": "object",
+  "additionalProperties": false,
+  "required": ["commits"],
+  "properties": {
+    "commits": {
+      "type": "object",
+      "additionalProperties": true
+    }
+  }
+}
+"""
+
+
+def passwd_lines_to_dict(lines):
+    """Take a list of passwd lines and produce a "name": "line" dictionary"""
+    res = dict()
+    for line in lines:
+        res[line.split(':')[0]] = line
+
+    return res
+
+
+def merge_passwd(primary, secondary):
+    """Merge lines from two passwd files
+
+    The lines from the "primary" file take precedence over the lines from
+    the "secondary" file meaning that if a user already exists in the first
+    one, the line will not be appended.
+
+    The result is stored in the primary file.
+    """
+    new_passwd_lines = []
+    with open(primary, "r") as p:
+        new_passwd_dict = passwd_lines_to_dict(p.readlines())
+        with open(secondary, "r") as s:
+            secondary_passwd_dict = passwd_lines_to_dict(s.readlines())
+            for name, passwd_line in secondary_passwd_dict.items():
+                if name not in new_passwd_dict:
+                    new_passwd_dict[name] = passwd_line
+
+            new_passwd_lines = list(new_passwd_dict.values())
```

Comment:
Could move that into the `writelines` call below and then don't even need the `new_passwd_lines` variable.